.buildkite/publish/publish-common.sh

```diff
@@ -18,11 +18,34 @@ source $SCRIPT_DIR/git-setup.sh
 VERSION_PATH="$PROJECT_ROOT/connectors/VERSION"
 export VERSION=$(cat $VERSION_PATH)
 
+# Variables for build.yaml
+version="${VERSION}"
+version_qualifier="${VERSION_QUALIFIER:-}"
+revision="$(git rev-parse HEAD)"
+repository="$(git config --get remote.origin.url)"
+
 if [[ "${USE_SNAPSHOT:-}" == "true" ]]; then
+  # SNAPSHOT workflows ignore version qualifier
   echo "Adding SNAPSHOT labeling"
   export VERSION="${VERSION}-SNAPSHOT"
+  version="${VERSION}"
+elif [[ -n "${VERSION_QUALIFIER:-}" ]]; then
+  # version qualifier should not append version in build.yaml
+  echo "Adding version qualifier labeling"
+  version="${VERSION}-${VERSION_QUALIFIER}"
 fi
 
+# Create a build.yaml file for reference after build process
+cat <<EOL > connectors/build.yaml
+version: "$version"
+qualifier: "$version_qualifier"
+revision: "$revision"
+repository: "$repository"
```

Comment:
Does this result in literal quote chars in the YAML file? If not, I suggest you add escaped quotes too, so that these do get quoted. Just to avoid any nasty bugs where a repository name has YAML special characters, or some shit.

Reply:
It results in literal quote chars. The file that gets build looks like this:

```
version: "9.0.0-beta1"
qualifier: "beta1"
revision: "27721f59f93e21980b81d5322248d2e6238b5aa2"
repository: "git@github.com:elastic/connectors-python.git"
```

Reply:
It also means empty values will look like this:

```
version: "9.0.0"
qualifier: ""
revision: "27721f59f93e21980b81d5322248d2e6238b5aa2"
repository: "git@github.com:elastic/connectors-python.git"
```

Reply:
fantastic